src/tiered/tiered_handle.c

```diff
@@ -361,10 +377,37 @@ __tiered_switch(WT_SESSION_IMPL *session, const char *config)
     tracking = false;
     WT_ERR(__wt_meta_track_off(session, true, ret != 0));
     WT_ERR(__tiered_update_dhandles(session, tiered));
+
+    if (tiered->current_id > 1 && tiered->current_id != prev_id) {
+        /*
+         * This code should be done in its own thread, as it may incur network delays or failure.
```

Comment:
I pushed a draft branch of the thread and work unit stuff. It is still very much in progress but I'll move this code into it when I write the function to process a flush work unit.